xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h

```diff
@@ -320,8 +329,14 @@ class ATTRIBUTE_HIDDEN CAddonBase
     return ADDON_STATUS_UNKNOWN;
   }
 
+  virtual ADDON_STATUS CreateInstanceEx(int instanceType, std::string instanceID, KODI_HANDLE instance, KODI_HANDLE& addonInstance, const char *version)
```

Comment:
You use this way that an addon can check itself to determine if it is compatible?

As far as OK, I would instead of "const char *" a "std :: string", this fits more to the C ++ interface and makes a string compare easier.